mfs/dir.go

```diff
@@ -131,7 +131,7 @@ func (d *Directory) cacheNode(name string, nd *dag.Node) (FSNode, error) {
 		ndir := NewDirectory(d.ctx, name, nd, d, d.dserv)
 		d.childDirs[name] = ndir
 		return ndir, nil
-	case ufspb.Data_File, ufspb.Data_Raw:
+	case ufspb.Data_File, ufspb.Data_Raw, ufspb.Data_Symlink:
```

Comment:
The reason this currently works (since we use mfs for add) is that we never instantiate a NewFile as a symlink (or when we do, in the current incorrect code, it never gets used).